File: javascript/bob/bob.js
var Bob = function() {}

Bob.prototype.hey = function(input) {
    if (input.match(/[A-Z]/) && input.isUpper()) { return "Whoa, chill out!"; }
    if (input.match(/\?$/)) { return "Sure."; }
    if (input.match(/^\s*$/)) { return "Fine. Be that way!"; }
    return "Whatever.";
}

String.prototype.isUpper = function() {
    return this.valueOf() === this.valueOf().toUpperCase();
}

module.exports = Bob;

File: ruby/bob/bob.rb
class Bob
    def self.hey(remark)
        remark.strip!
        return "Whoa, chill out!" if remark =~ /[a-zA-Z]+/ && remark == remark.upcase
        return "Sure." if remark =~ /\?\Z/
        return "Fine. Be that way!" if remark =~ /\A[\W]*\Z/
        "Whatever."
    end
end
module BookKeeping
    VERSION = 1
end